### PR TITLE
Move the quantization API to OSS

### PR DESCRIPTION
**Pull request by mcremon-meta**
Summary: As titled.

Differential Revision: D58680953

### CODE REVIEW

File: backends/cadence/aot/compiler.py [hunk 1]
```diff
@@ -16,18 +18,56 @@
...
 from executorch.exir import EdgeCompileConfig, EdgeProgramManager, to_edge
+from torch._export import capture_pre_autograd_graph
 from torch.ao.quantization.pt2e.export_utils import model_is_exported
+from torch.ao.quantization.quantize_pt2e import convert_pt2e, prepare_pt2e
 
 from torch.export import export
 from torch.export.exported_program import ExportedProgram
 
+# pyre-ignore[33]: `InputsArgsType` cannot alias to `Any`.
+InputsArgsType = Tuple[Any, ...]
```
Comment (dulinriley) [inline backends/cadence/aot/compiler.py:33]: Perhaps `object` instead to fix this error message?
```suggestion
InputsArgsType = Tuple[object, ...]
```

File: backends/cadence/aot/compiler.py [hunk 2]
```diff
@@ -16,18 +18,56 @@
...
+    quantizer = CadenceQuantizer()
+
+    # Export with dynamo
+    model_exp = capture_pre_autograd_graph(model, inputs)
+
+    # Prepare
+    prepared_model = prepare_pt2e(model_exp, quantizer)
+
+    # Calibrate
+    prepared_model(*inputs)
```
Comment (dulinriley) [inline backends/cadence/aot/compiler.py:54]: You'll need to calibrate with more than one sample normally for good PTQ accuracy. Is this API meant more for testing?

**Review by dulinriley — APPROVED** [review]

File: backends/cadence/aot/compiler.py (hunk 2 above)
Comment (mcremon-meta) [reply backends/cadence/aot/compiler.py:54]: yup this is all dummy for now. We don't even look at quantization accuracy, only implementation accuracy (ref outputs are from the converted forward model).
